plugins/inputs/fritzbox/fritzbox.go

```diff
@@ -0,0 +1,454 @@
+	fields["init_errors"] = statisticsTotal.NewInitErrors
+	fields["init_timeouts"] = statisticsTotal.NewInitTimeouts
+	fields["loss_of_framing"] = statisticsTotal.NewLossOfFraming
+	fields["errored_secs"] = statisticsTotal.NewErroredSecs
+	fields["severly_errored_secs"] = statisticsTotal.NewSeverelyErroredSecs
+	fields["fec_errors"] = statisticsTotal.NewFECErrors
+	fields["atuc_fec_errors"] = statisticsTotal.NewATUCFECErrors
+	fields["hec_errors"] = statisticsTotal.NewHECErrors
+	fields["atuc_hec_errors"] = statisticsTotal.NewATUCHECErrors
+	fields["crc_errors"] = statisticsTotal.NewCRCErrors
+	fields["atuc_crc_errors"] = statisticsTotal.NewATUCCRCErrors
+	acc.AddFields("fritzbox_dsl", fields, tags)
+	if plugin.Log.Level().Includes(telegraf.Debug) {
+		plugin.logMetric("fritzbox_dsl", tags, fields)
+	}
+	return nil
+}
+
+func (plugin *Fritzbox) gatherWlanInfo(acc telegraf.Accumulator, deviceClient *tr064.Client, service tr064.ServiceDescriptor) error {
+	serviceClient := wlanconfig.ServiceClient{
+		TR064Client: deviceClient,
+		Service:     service,
+	}
+	info := &wlanconfig.GetInfoResponse{}
+	err := serviceClient.GetInfo(info)
+	if err != nil {
+		return err
+	}
+	if info.NewStatus != "Up" {
+		return nil
+	}
+	totalAssociations := &wlanconfig.GetTotalAssociationsResponse{}
+	err = serviceClient.GetTotalAssociations(totalAssociations)
+	if err != nil {
+		return err
+	}
+	tags := make(map[string]string)
+	tags["source"] = serviceClient.TR064Client.DeviceUrl.Hostname()
+	tags["service"] = serviceClient.Service.ShortId()
+	tags["ssid"] = info.NewSSID
+	tags["channel"] = strconv.Itoa(int(info.NewChannel))
+	tags["band"] = plugin.wlanBandFromInfo(info)
+	fields := make(map[string]interface{})
+	fields["total_associations"] = totalAssociations.NewTotalAssociations
+	acc.AddGauge("fritzbox_wlan", fields, tags)
+	if plugin.Log.Level().Includes(telegraf.Debug) {
+		plugin.logMetric("fritzbox_wlan", tags, fields)
+	}
+	return nil
+}
+
+func (plugin *Fritzbox) wlanBandFromInfo(info *wlanconfig.GetInfoResponse) string {
+	band := info.NewX_AVM_DE_FrequencyBand
+	if band != "" {
+		return band
+	}
+	if 1 <= info.NewChannel && info.NewChannel <= 14 {
+		return "2400"
+	}
+	return "5000"
+}
+
+func (plugin *Fritzbox) gatherHostsInfo(acc telegraf.Accumulator, deviceClient *tr064.Client, service tr064.ServiceDescriptor) error {
+	if plugin.queryCycleCounter != 0 {
+		return nil
+	}
+	serviceClient := hosts.ServiceClient{
+		TR064Client: deviceClient,
+		Service:     service,
+	}
+	connections, err := plugin.fetchHostsConnections(&serviceClient)
+	if err != nil {
+		return err
+	}
+	for _, connection := range connections {
+		_, err = uuid.Parse(connection.RightDeviceName)
+		if err == nil {
+			continue
+		}
+		tags := make(map[string]string)
+		tags["source"] = serviceClient.TR064Client.DeviceUrl.Hostname()
+		tags["service"] = serviceClient.Service.ShortId()
+		tags["host"] = connection.RightDeviceName
```

Comment:
`host` conflicts with the tag telegraf uses to fill the hostname where the telegraf agent is running.

Reply:
Good spotted. I rename it to node which actually matches the corresponding mesh terminology in this context quite well.